src/client/lixa_tx.c

```diff
@@ -53,6 +53,69 @@
 #endif /* LIXA_TRACE_MODULE */
 #define LIXA_TRACE_MODULE   LIXA_TRACE_MOD_CLIENT_TX
 
+static char *tx_open_profile = NULL;
+static GMutex profile_mutex = G_STATIC_MUTEX_INIT;
+
+int lixa_tx_set_profile(int *txrc, const char *profile)
+{
+    enum Exception {
+        G_STRDUP_ERROR,
+        NONE
+    } excp;
+    int ret_cod = LIXA_RC_INTERNAL_ERROR;
+    *txrc = TX_FAIL;
+
+    LIXA_TRACE_INIT;
+    LIXA_CRASH_INIT;
+    LIXA_TRACE(("lixa_tx_set_profile\n"));
+
+    g_mutex_lock( &profile_mutex );
+    TRY {
+        if (tx_open_profile) {
+            free(g_steal_pointer(&tx_open_profile));
+        }
+        if (profile) {
+	    tx_open_profile = g_strdup(profile);
+            if (tx_open_profile == NULL) {
+                THROW( G_STRDUP_ERROR );
+            } else {
+                THROW( NONE );
+            }
+        } else {
+            THROW( NONE );
+        }
+    } CATCH {
+        switch (excp) {
+        default:
```

Comment:
I have never seen a "default:" as first exit point in a switch. I don't know how it behaves: please put as the last one as in all LIXA functions:

                default:
        	        ret_cod = LIXA_RC_INTERNAL_ERROR;
	                break;